From 408bb69874e1488f3f294bb461da6042feda80ea Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Thu, 2 Nov 2017 08:01:54 +0000
Subject: avoid trying to resolve a filename unnecessarily

-use a regular expression over trivial string comparison to
determine whether a potential filename contains a Quod Libet
placeholder (e.g. <title>)
---
 quodlibet/quodlibet/formats/_audio.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/quodlibet/quodlibet/formats/_audio.py b/quodlibet/quodlibet/formats/_audio.py
index 63d794ff7..57b7856c5 100644
--- a/quodlibet/quodlibet/formats/_audio.py
+++ b/quodlibet/quodlibet/formats/_audio.py
@@ -11,6 +11,7 @@
 # more readable, unless they're also faster.
 
 import os
+import re
 import shutil
 import time
 from collections import OrderedDict
@@ -586,9 +587,10 @@ class AudioFile(dict, ImageContainer):
         #print_d("looking for lyrics files:\n%s" % '\n'.join(pathfiles.keys()))
 
         match_ = ""
+        rx_params = re.compile('[^\\\]<[^' + re.escape(os.sep) + ']*[^\\\]>')
         for pathfile in pathfiles.keys():
             path = expanduser(pathfile)
-            if '<' in path:
+            if rx_params.search(path):
                 path = ArbitraryExtensionFileFromPattern(pathfile).format(self)
             pathfiles_expanded[path] = path  # resolved as late as possible
             if os.path.exists(path):
-- 
2.14.1

